### PR TITLE
* Support ssdlite

### PR DESCRIPTION
**Pull request by tmyapple**
I tried to compare the mobilenet-ssd in gluon to the one published in tensorflow object detection api.
Gluon-cv model which is one-to-one match to what is being described in (https://arxiv.org/abs/1512.02325) has ~23M weights, while TF arch has only 6.7M.
Main differences between the models:
1. Number of anchors per branch - in the lighter model [3,6,6,6,6,6] (vs. original [4,6,6,6,4,4]) - number of proposals 1917 vs 2268 .
2. prediction and regression head in the lighter model has pointwise convolutions(1x1 kernels), vs the original 3x3 kernels.
3. mindepth is 64, and there is a reduce ratio of 0.5 - it reduces the number of model weights a lot
4. The first branch - doesn't has the the complementary anchor which is 1:1 * (geomtric mean of current size and next branch size). And its first anchor has a size which is 10% of the image.

### CODE REVIEW

File: gluoncv/model_zoo/ssd/anchor.py [hunk 1]
```diff
@@ -50,14 +51,21 @@ def _generate_anchors(self, sizes, ratios, step, alloc_size, offsets):
...
+                    anchors.append([cx, cy, sizes[0], sizes[0]])
+                else:
+                    anchors.append([cx, cy, sizes[0] / 2, sizes[0] / 2])
                 # size = sizes[0], ratio = ...
                 for r in ratios[1:]:
                     sr = np.sqrt(r)
                     w = sizes[0] * sr
                     h = sizes[0] / sr
                     anchors.append([cx, cy, w, h])
+                if not self._is_lite:
```
[the remark this answers is not in the document]
Comment (tmyapple) [reply gluoncv/model_zoo/ssd/anchor.py]: changed it to the original order

**Comment by tmyapple** [comment]
@zhreshold , thanks for the comments,
1&2 Sounds good , i will try it out
3. training got to 19.2mAP. I guess that some optimization can be made with the training process, I've used the arguments given to ssd_512_mobilenet1.0_coco, without seraching for better training flow(epochs/lr/optimizer).

**Comment by zhreshold** [comment]
@tmyapple Thanks, 19.2 map seems to be reasonable and better than tf version

**Comment by tmyapple** [comment]
I've pushed some changes. As suggested by @zhreshold 
Main changes:
- Created a subclass for LiteAnchorGenerator
- SSD default is the original SSDAnchorGenerator

File: gluoncv/model_zoo/ssd/ssd.py [hunk 2]
```diff
@@ -397,8 +406,8 @@ def get_ssd(name, base_size, features, filters, sizes, ratios, steps, classes,
     pretrained_base = False if pretrained else pretrained_base
     base_name = None if callable(features) else name
     net = SSD(base_name, base_size, features, filters, sizes, ratios, steps,
-              pretrained=pretrained_base, classes=classes, ctx=ctx, root=root,
-              minimal_opset=pretrained, **kwargs)
```
Comment (zhreshold) [inline gluoncv/model_zoo/ssd/ssd.py:401]: `minimal_opset=pretrained` deleted by mistake?

**Comment by zhreshold** [comment]
@tmyapple Thanks! LGTM now. Can you help fix the lint error so that the CI can pass?

**Comment by tmyapple** [comment]
* hopefully lint will pass

**Comment by mli** [comment]
Job PR-1252-10 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1252/10/index.html 
Code coverage of this PR:  vs. Master:

**Comment by zhreshold** [comment]
@tmyapple thanks for your great contribution, this is merged!

**Comment by tmyapple** [comment]
thanks, it is a pleasure to take part and contribute...

**Pull request merged**